Co-authored-by: zainmfjavaid <zainmfjavaid@users.noreply.github.com>

Ariya3 opened the pull request.

Ariya3 committed:
Co-authored-by: zainmfjavaid <zainmfjavaid@users.noreply.github.com>

azvanderpas commented on src/main/java/frc/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,45 @@
+  private CANSparkMax m_motorController1 = new CANSparkMax(Constants.kMotorOneID, CANSparkMaxLowLevel.MotorType.kBrushless);
+  private CANSparkMax m_motorController2 = new CANSparkMax(Constants.kMotorTwoID, CANSparkMaxLowLevel.MotorType.kBrushless);
+  
+  public IntakeSubsystem(boolean reverse, boolean on){
```
Turning on the motors shouldn't be done in the constructor. Make a new method for commands to use and leave configuration steps in the constructor.

sarango2158 replied:
Same comment as for Hopper -- you should create a method to start, stop, or change the direction.

sarango2158 commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -12,4 +12,9 @@
  */
-public final class Constants {}
+public final class Constants {
+    public static final int kMotorOneID = 11;
```
Remember all of the subsystems will use the Constants file. How can we name the motors that belong to intake so they're easy to identify?

azvanderpas replied:
Maybe add a TODO comment to rename these constants and ID numbering once the mechanisms have been designed.

azvanderpas commented on src/main/java/frc/robot/subsystems/HopperSubsystem.java:
```diff
@@ -0,0 +1,34 @@
+  // here. Call these from Commands.
+  private CANSparkMax m_motorController3 = new CANSparkMax(Constants.kMotorThreeID, CANSparkMaxLowLevel.MotorType.kBrushless);
+  
+  public HopperSubsystem(boolean on){
```
Same comment as on intake. Control motor speed in a different method

sarango2158 replied:
This is your constructor, it is called just once when the HopperSubsystem is created (using "new"). It is used to assign values to any of the object's data and potentially set an initial state. If you want a way to control whether the Hopper is running at a given time, you should create a new method for that purpose.

azvanderpas reviewed (changes requested):
Good start. Don't forget to create a command class and hook them up to buttons as well.

Ariya3 committed:
Co-authored-by: zainmfjavaid <zainmfjavaid@users.noreply.github.com>

zainmfjavaid committed:
Update Constants.java

zainmfjavaid committed:
Update HopperSubsystem.java

Added a shooter motor controller. To have an effective CDC, we have to base our timing on whether the shooter is on and what speed it's rotating at.

zainmfjavaid committed:
Update Robot.java

azvanderpas commented on src/main/java/frc/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,55 @@
+  
+  public void IntakeSwitch(boolean on){
+    if (on){
+      m_motorController1.set(Constants.kIntakeMotorSpeed);
```
Don't forget about the [follow function](https://codedocs.revrobotics.com/java/com/revrobotics/cansparkmax#follow(com.revrobotics.CANSparkMax)). This can be used to make one CANSparkMax always do the same thing as another. If this code becomes more complex it can save you a lot of copy/pasting

azvanderpas commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -16,8 +19,9 @@
   private Command m_autonomousCommand;
-
   private RobotContainer m_robotContainer;
+  private IntakeSubsystem m_intakeSubsystem;
```
It would be better to instantiate the subsystems in RobotContainer because you'll want to use them with your Command classes and button bindings

Ariya3 committed:
Co-authored-by: zainmfjavaid <zainmfjavaid@users.noreply.github.com>
Co-authored-by: Noga Rosenberg <NogaRoze@users.noreply.github.com>

Ariya3 committed:
Co-authored-by: zainmfjavaid <zainmfjavaid@users.noreply.github.com>
Co-authored-by: Noga Rosenberg <NogaRoze@users.noreply.github.com>

Ariya3 committed:
hi

Ariya3 committed:
Co-authored-by: zainmfjavaid <zainmfjavaid@users.noreply.github.com>

zainmfjavaid committed:
Update Constants.java

Ariya3 committed:
hi

zainmfjavaid committed:
Update Constants.java

added bumper button constants

Ariya3 committed:
Merge branch 'intake' of https://github.com/AusTINCANsProgrammingTeam/2022RobotCode into intake

zainmfjavaid committed:
Update RobotContainer.java

added basic button mappings for forward and reverse intake commands

azvanderpas commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -34,15 +44,20 @@ public RobotContainer() {
-  private void configureButtonBindings() {}
+  private void configureButtonBindings() {
+    // TODO: figure out if we want to use whenPressed or WhileHeld
+    mButtons[Constants.kLeftBumperButton].whenPressed(mIntakeForwardCommand);
```
Each index of mButtons (or at least the ones you want to use) needs to get initialized prior to this. Currently this will cause a runtime error.

azvanderpas commented on src/main/java/frc/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,44 @@
+
+  public void ForwardIntake(){
+    m_intakeMotorController1.setInverted(false);
+    m_intakeMotorController2.setInverted(false);
```
m_intakeMotorController2 references here and in ReverseIntake should be removed since its already setup to follow m_intakeMotorController1

azvanderpas commented on src/main/java/frc/robot/subsystems/HopperSubsystem.java:
```diff
@@ -0,0 +1,35 @@
+  // Put methods for controlling this subsystem
+  // here. Call these from Commands.
+  private CANSparkMax m_hopperController = new CANSparkMax(Constants.kHopperMotorThreeID, CANSparkMaxLowLevel.MotorType.kBrushless);
+  private CANSparkMax m_shoooterMotorController = new CANSparkMax(Constants.kShooterMotorID, CANSparkMaxLowLevel.MotorType.kBrushless);
```
The shooter motor controller should only be instantiated by the Shooter subsystem. If you want to get the speed of the shooter motor, then a function that returns that must be provided by the shooter subsystem class. I'll try to cover this better at today's meeting.

azvanderpas commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -34,15 +44,20 @@ public RobotContainer() {
    */
-  private void configureButtonBindings() {}
+  private void configureButtonBindings() {
+    // TODO: figure out if we want to use whenPressed or WhileHeld
```
The way your command classes are written, whileHeld makes more sense. Take a look these wpilib docs to get a better understanding:
[Binding Commands To Triggers](https://docs.wpilib.org/en/stable/docs/software/commandbased/binding-commands-to-triggers.html)
[The Scheduler Run Sequence](https://docs.wpilib.org/en/stable/docs/software/commandbased/command-scheduler.html#the-scheduler-run-sequence)

zainmfjavaid committed:
Update Constants.java

added new button for the hopper and comments

azvanderpas commented on src/main/java/frc/robot/commands/IntakeReverseCommand.java:
```diff
@@ -0,0 +1,40 @@
+  // Returns true when the command should end.
+  @Override
+  public boolean isFinished() {
+    return false;
```
Note that if you want to use whenPressed, isFinished will need to return true in some case (or some other action causes the Command to be canceled) in order for the command to end. As written, this command will start but never call the end function. See my other comment for wpilib docs to review.

azvanderpas commented on src/main/java/frc/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,44 @@
+
+
+/** Add your docs here. */
+public class IntakeSubsystem extends SubsystemBase {
```
This class is missing a constructor

azvanderpas commented on src/main/java/frc/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -0,0 +1,44 @@
+  }
+  
+  public void IntakeSwitch(boolean on){
+    m_intakeMotorController2.follow(m_intakeMotorController1, false);
```
Move to constructor method

zainmfjavaid committed:
Update HopperSubsystem.java

zainmfjavaid committed:
Create HopperCommand.java

zainmfjavaid committed:
Update HopperCommand.java

zainmfjavaid committed:
Update HopperCommand.java

zainmfjavaid committed:
Update RobotContainer.java

mapped A button to elevator

zainmfjavaid committed:
Update HopperCommand.java

zainmfjavaid committed:
Update Constants.java

Removed shooter id

zainmfjavaid committed:
Update IntakeSubsystem.java

Removed motorController2 references to reflect follow

Ariya3 committed:
Merge branch 'intake' of https://github.com/AusTINCANsProgrammingTeam/2022RobotCode into intake

Ariya3 committed:
Merge branch 'intake' of https://github.com/AusTINCANsProgrammingTeam/2022RobotCode into intake

Co-authored-by: zainmfjavaid <zainmfj@gmail.com>

Ariya3 committed:
hi

Ariya3 committed:
addressed comments
